docs/reference/integrations/bigquery.rst

```diff
@@ -4,41 +4,65 @@
 BigQuery
 ##############
 
-To add a BigQuery datasource do this:
+How to set up BigQuery as a DataSource
+======================================
+
+To add BigQuery datasource do this:
+
+Follow the `Google Cloud library guide <https://googleapis.dev/python/google-api-core/latest/auth.html>`_
+for authentication.
+
+Install the pybigquery package for the BigQuery sqlalchemy dialect (``pip install pybigquery``)
+
+
+Use the CLI:
 
 1. Run ``great_expectations datasource new``
-2. Choose the *SQL* option from the menu.
-3. When asked which SQLAlchemy driver to use, enter ``other``.
-4. Consult the `PyBigQuery <https://github.com/mxmzdlv/pybigquery`_ docs
+2. Choose "Big Query" from the list of database engines, when prompted.
+3. Consult the `PyBigQuery <https://github.com/mxmzdlv/pybigquery`_ docs
    for help building a connection string for your BigQuery cluster. It will look
    something like this:
 
     .. code-block:: python
 
         "bigquery://project-name"
 
+   If you want GE to connect to one of the Google's public datasets, the URL should be:
+
+    .. code-block:: python
+
+        "bigquery://project-name/bigquery-public-data"
 
-5. Paste in this connection string and finish out the interactive prompts.
+5. Paste in this connection string when prompted for SQLAlchemy URL and finish out the interactive prompts.
 6. Should you need to modify your connection string you can manually edit the
    ``great_expectations/uncommitted/config_variables.yml`` file.
 
-Custom Queries with SQL datasource
-==================================
 
-While other backends use temporary tables to generate batches of data from
-custom queries, BigQuery does not support ephemeral temporary tables. As a
-work-around, GE will create or replace a *permanent table* when the user supplies
-a custom query.
+Note: environment variables can be used to store the SQLAlchemy URL instead of the file, if preferred - search documentation for "Managing Environment and Secrets".
 
-Users can specify a table via a Batch Kwarg called ``bigquery_temp_table``:
+How to specify a query result as a batch
+==========================================
+
+When you specify a result set of a query as a batch to be validated, GE executes the query and stores
+the result in a temporary table in order to optimize the performance of validation.
+
+BigQuery does not support ephemeral temporary tables. As a
+work-around, GE will create or replace a *permanent table*.
+
+The user must provide the name of the table via an additional key when constructing
+BatchKwargs for the query: ``bigquery_temp_table``.
+
+Here is an example:
 
     .. code-block:: python
 
         batch_kwargs = {
             "query": "SELECT * FROM `my-project.my_dataset.my_table`",
-            "bigquery_temp_table": "my_other_dataset.temp_table"
+            "bigquery_temp_table": "my-project.my_other_dataset.temp_table"
```

Comment:
consistency: `project-name` instead of `my-project`

Reply:
Addressed.